src/comparable.ts

```diff
@@ -0,0 +1,37 @@
+export default interface Comparable<T extends Comparable<T>> {
```

Comment:
comparable type extends comparable? What does this do?

Reply:
The item you compare to also implements the Comparable interface.

Reply:
Hmm this seems weird and maybe too complex? Not sure if there is a better way though. Anyway

Reply:
We basically need to do it that way to properly implement the sorting methods. When you want to sort a `Comparable<T>[]` array, you will need to provide a callback function that takes two `Comparable<T>`s as argument. By doing it this way, we can just call the `compareTo` function. 

Maybe there is a cleaner way to accomplish this, but it was the first one I could think of.